File: eAutobus.Mobile/eautobusmobile/lib/models/karta/Karta.dart
import 'dart:core';

import 'package:json_annotation/json_annotation.dart';

part 'Karta.g.dart';

@JsonSerializable()
class Karta {
  int? kartaId;
  String? nacinPlacanja;
  int? vrstaKarteID;
  int? tipKarteID;
  int? odredisteID;
  int? polazisteID;
  String? cijenaPrikaz;
  int? cijena;
  String? ime;
  String? prezime;
  DateTime? datumVadjenjaKarte;
  DateTime? datumVazenjaKarte;
  int? kupacID;
  String? korisnickoIme;

  Karta() {}
  factory Karta.fromJson(Map<String, dynamic> json) => _$KartaFromJson(json);

  /// Connect the generated [_$KartaToJson] function to the `toJson` method.
  Map<String, dynamic> toJson() => _$KartaToJson(this);
}

File: eAutobus/Database/Karta.cs
﻿namespace eAutobus.Database
{
    public class Karta
    {
        public int KartaID { get; set; }


        public List<KartaKupac> KupacList { get; set; }

        public TipKarte TipKarte { get; set; }
        public int TipKarteID { get; set; }
        public int PolazisteID { get; set; }
        public Stanica Polaziste { get; set; }
        public int OdredisteID { get; set; }
        public Stanica Odrediste { get; set; }
        public int VrstaKarteID { get; set; }
        public bool IsDeleted { get; set; }

        public VrstaKarte VrstaKarte { get; set; }
        public double Cijena { get; set; }
        public string NacinPlacanja { get; set; }
        public List<PlatiKartu> PlaceneKarte { get; set; }
    }
}
